recipes/libpng/all/conanfile.py

```diff
@@ -72,48 +71,24 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            try:
-                del self.options.fPIC
-            except Exception:
-                pass
-        try:
-            del self.settings.compiler.libcxx
-        except Exception:
-            pass
-        try:
-            del self.settings.compiler.cppstd
-        except Exception:
-            pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("zlib/1.2.13")
 
+    def validate(self):
+        if Version(self.version) < "1.6" and self.info.settings.arch == "armv8" and is_apple_os(self):
+            raise ConanInvalidConfiguration(f"{self.ref} could not be cross build on Mac.")
```

Comment:
Can this be worded slightly better? This doesn't actually have to do with cross-compiling does it? It looks like this doesn't support building for `armv8` for Apple systems, whether or not that's native right?

Reply:
I'd suggest a move positive language here too

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently does not support building with {self.info.settings.arch}. Please contribute.")
```

Reply:
I would say "does not support building for"

Reply:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently does not building for {self.info.settings.os} {self.info.settings.arch}. Contributions are welcomed")
```